model/map-jpa/src/main/java/org/keycloak/models/map/storage/jpa/group/JpaGroupModelCriteriaBuilder.java

```diff
@@ -81,6 +81,7 @@ public JpaGroupModelCriteriaBuilder compare(SearchableModelField<? super GroupMo
                         Join<JpaGroupEntity, JpaGroupAttributeEntity> join = root.join("attributes", JoinType.LEFT);
                         return cb.and(
                             cb.equal(join.get("name"), value[0]),
+                            cb.equal(join.get("value_hash"), sha256((String) value[1])),
```

Comment:
Question: is it needed to include the value in the comparison? A different value would lead to a different hash, wouldn't it? So it should be enough to compare the name and hash. WDYT?

Reply:
Good question. I believe that set of possible hashes is smaller than set of possible values. In other words I think it may happen (although it is very low probability) that two different values could have the same hash.  Sha256 hashes have a finite length, but there is infinite number of possible characters.

Reply:
I believe the odds of a sha256 collision are very very low, but I'm not really against comparing values. Just wondered if we could do without this. @hmlnarik what is your take on this?

Reply:
Thinking about it more ... we shouldn't be able to achieve a state where two different values would have a same hash due to the unique constraint we've introduced (at least for postgres). 

So it seems to me that we could omit the value itself from the query and let there just value_hash. 

@hmlnarik wdyt?

Reply:
> Question: is it needed to include the value in the comparison? A different value would lead to a different hash, wouldn't it? So it should be enough to compare the name and hash. WDYT?

The value should be kept in the query condition as a security measure. If there was a collision attack against SHA256 in the future, this could stop the attacker.

Reply:
In that case I'm approving this PR as I have no further comments.